src/products/components/ProductVariantCreatePage/form.tsx

```diff
@@ -185,10 +193,13 @@ function useProductVariantCreateForm(
   } = useMetadataChangeTrigger();
 
   const changeMetadata = makeMetadataChangeHandler(handleChange);
-  const handleAttributeChange = createAttributeChangeHandler(
-    attributes.change,
-    triggerChange,
-  );
+
+  const handleAttributeChangeWithName = (id: string, value: string) => {
+    triggerChange();
+    attributes.change(id, value === "" ? [] : [value]);
+    handleChange({ target: { value, name: "name" } });
```

Comment:
just to comment: don't hesitate to refactor big components. You can spot many of them in our codebase, but they can be decoupled into smaller units where one component/hook means one responsibility. If you feel you can do it, please do - it always makes code more testable and readable